test/e2e/storage/volume_metrics.go

```diff
@@ -165,10 +155,26 @@ var _ = SIGDescribe("[Serial] Volume metrics", func() {
 			kubeletmetrics.VolumeStatsInodesUsedKey,
 		}
 
-		for _, key := range volumeStatKeys {
-			kubeletKeyName := fmt.Sprintf("%s_%s", kubeletmetrics.KubeletSubsystem, key)
-			verifyVolumeStatMetric(kubeletKeyName, pvc.Namespace, pvc.Name, kubeMetrics)
-		}
+		waitErr := wait.Poll(30*time.Second, 5*time.Minute, func() (bool, error) {
+			// Grab kubelet metrics from the node the pod was scheduled on
+			kubeMetrics, err := metricsGrabber.GrabFromKubelet(pod.Spec.NodeName)
+			if err != nil {
+				framework.Logf("Error fetching kubelet metrics")
+				return false, err
+			}
+
+			for _, key := range volumeStatKeys {
```

Comment:
Is there a reason this `for` loop is inside the  `Poll` call?

Reply:
You're right - doesn't need to be. Moved out - we use the poll simply to determine if the metrics have been updated and then outside the `Poll` call go validate